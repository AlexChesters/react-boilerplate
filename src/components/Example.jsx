import React, { Component } from 'react'

export default class Example extends Component {
  static get defaultProps () {
    return {
      text: 'Hello, world!',
      style: {
        display: 'flex',
        justifyContent: 'center'
      }
    }
  }
  render () {
    return (
      <section style={this.props.style}>
        <h1>{this.props.text}</h1>
      </section>
    )
  }
}
