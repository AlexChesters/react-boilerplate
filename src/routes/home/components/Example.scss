@import '../../../styles/colors';

.example {
  color: map-get($colors, 'text');
}
