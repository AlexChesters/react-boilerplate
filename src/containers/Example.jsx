import React, { Component } from 'react'
import { connect } from 'react-redux'

import * as actionCreators from '../actions/action-creators.js'

import Example from '../components/Example'

export class ExampleContainer extends Component {
  render () {
    this.props.getRandomString()
    return <Example text={this.props.randomString} />
  }
}

const mapStateToProps = state => {
  return {
    randomString: state.get('randomString')
  }
}
export default connect(mapStateToProps, actionCreators)(ExampleContainer)
