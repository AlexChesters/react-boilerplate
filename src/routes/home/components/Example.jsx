import React, { Component } from 'react'
import PropTypes from 'prop-types'

import './Example.scss'

export default class Example extends Component {
  static get defaultProps () {
    return {
      text: 'Hello, world!'
    }
  }
  render () {
    return (
      <section className='example'>
        <h1>{this.props.text}</h1>
      </section>
    )
  }
}

Example.propTypes = {
  text: PropTypes.string
}
