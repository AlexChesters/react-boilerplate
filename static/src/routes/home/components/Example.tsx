import React from 'react'
import PropTypes from 'prop-types'

import './Example.scss'

type Props = {
  title: string,
  subtitle: string
}

const Example = (props: Props) => {
  return (
    <section className='example'>
      <h1>{props.title}</h1>
      <h4>{props.subtitle}</h4>
    </section>
  )
}

Example.propTypes = {
  title: PropTypes.string,
  subtitle: PropTypes.string
}

export default Example
