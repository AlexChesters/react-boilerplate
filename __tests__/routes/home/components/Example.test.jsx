/* eslint-env jest */

import React from 'react'
import renderer from 'react-test-renderer'

import Example from '../../../../src/routes/home/components/Example'

describe('Example', () => {
  it('renders correctly', () => {
    const tree = renderer.create(
      <Example path={'/2017-04'} text='2017-04' />
    ).toJSON()
    expect(tree).toMatchSnapshot()
  })
})
