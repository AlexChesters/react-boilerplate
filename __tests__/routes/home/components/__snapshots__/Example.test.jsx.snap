// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Example renders correctly 1`] = `
<section
  className="example"
>
  <h1>
    2017-04
  </h1>
</section>
`;
